### ISSUE TITLE
"Sign in with Google" button fails when third-party cookies are blocked

### ISSUE THREAD

**Issue by alkema** [metabase/metabase#6741]
### Signing in while using Chrome on OS X with the "Sign in with Google" button fails. 

How do I know Google Apps and Metabase are configured correctly for this? Because it works on all other OS's and browser combinations I have tried.

**Details for failing case:**

- Browser and version: Chrome Version 63.0.3239.132 (Official Build) (64-bit)
- Operating System: OS X Sierra 10.12.6
- Databases: MySQL
- Metabase version: You're on version v0.27.2 (Built on 2017-12-12)
- Metabase hosting environment: Happens on both Heroku and OS X locally (`java -jar metabase.jar`)
- Metabase internal database: Postgres
- Metabase relevant config: `GOOGLE SIGN-IN` turned on with custom domain under "Allow users to sign up on their own if their Google account email address is from:"
- *Repeatable steps to reproduce the issue*

Note the below "Expected behavior" works fine on:

* OS X : Firefox and Safari
* iOS (iPhone 7) : Chrome, Safari

**Expected behavior:**

1. Go to login : `/auth/login` on http://localhost:3000 or production domain.
2. Press "Sign in with Google" button.
3. Google account chooser pops up, choose account and or sign in to Google.
4. I am logged in.

**Actual behavior:**

1. Go to login : `/auth/login` on http://localhost:3000 or production domain.
2. Press "Sign in with Google" button.
3. Google account chooser pops up, choose account and or sign in to Google.
4. I am *not* logged in, and there is a JavaScript error in Chrome's console: 

```
There was an error logging in {error: "popup_closed_by_user"}
```

Entire log from Chrome console [attached](https://github.com/metabase/metabase/files/1653317/Chrome-Version-63.0.3239.132-Official-Build-64-bit.log).

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by satyadeepk** [comment]
This issue happened to us on 0.28.3. 
But after updating to 0.28.4, this doesn't seem to happen again. 
@alkema Probably you can try upgrading to latest and see.

**Comment by amit** [comment]
I have the same issue. But it worked for me when I enabled third party cookies. Not sure if metabase could use a non-js based flow for Google OAuth (using redirect URLs). I would rather not enable third-party cookies nor allow google.com as an exception (especially google ;) )

**Comment by camsaul** [comment]
We could consider trying to use a non-JS flow for Google Auth (PRs welcome!), but in the meantime it sounds like you might have to enable third-party google.com cookies on Metabase in order to use Google Auth. @alkema did upgrading to the latest version and/or enabling third-party cookies solve your issue?

**Comment by bramswenson** [comment]
Works for me in v0.29.0, but only with "Privacy Badger" Chrome extension disabled.

**Comment by soobrosa** [comment]
Works for me in _Safari_.

Chrome definitely does not work.

Firefox might, but if you already have an account with an other domain you are signed in this one overrides.

**Comment by vbalazs** [comment]
I did some experiments:
* In Firefox (61.0.1) only works if "Accept third-party cookies and site data" set to "always"
* In Chrome (68.0.3440.84) it works even if I have "Block third-party cookies" set (<tinfoil hat ON> maybe chrome always trusts Google regardless of this option? haha)
* in Safari I could not find such a fine grained cookie configuration so it probably works because it always accepts 3rd party cookies if accepting cookies are enabled in general

**Comment by davidstoker** [comment]
I had a user of mine encounter this and did some experiments in Chrome 79.0.3945.117. If the "Block third-party cookies" is on, login with Google does not work and throws the `popup_closed_by_user` error. Toggle that option off, open a new tab and login works.

**Comment by radujica** [comment]
So think this issue is still valid. We ran into this, though maybe also due to our more complex setup. Two valid scenarios we found for Chrome were:

- use the "block third-party cookies" option and add a metabase rule/site to always allow cookies (ticking the "including third-party cookies on this site"); this can be added from the eye option in the right of the url in Chrome
- use the "block third-party cookies in incognito" option and nothing else to do

**Comment by flamber** [comment]
Fixed by #22868 in the upcoming version 0.44

**Comment by soobrosa** [comment]
❤️
